Find intersect elems

MengnanLi91 opened the pull request:
<!--
If this PR is associated with an issue be sure to reference it
by including "refs #<issue>" or "closes #<issue>" (e.g., #closes #1234).

If this PR implements an enhancement that does not have an existing issue, please add the following:
-->
The PR builds a helper utility to computes the intersection elements mapping between overlay mesh and main mesh.

## Reason
<!--Why do you need this feature or what is the enhancement?-->
This is an intermedia step for doing radiative heat transfer between unmeshed gap region 

## Design
<!--A concise description (design) of the enhancement.-->
Loop over the main mesh and overlay mesh to determine whether the elements are intersected and store those intersected ones in a map. The overlay mesh generator PR is brought over from #23513 since it hasn't been not merged yet.

<!--
If this PR implements a bug fix, please create an issue for the bug and reference the issue.
-->
Intermedia step for #22493

MengnanLi91 committed:
Basic function of intermediate mesh storage #23486

Add test case and doco

Update MeshGenerator.C

Update tests

Update MeshGenerator.md

Add gold mesh files

Update tests

Update tests

MengnanLi91 committed:
Basic function of intermediate mesh storage

MengnanLi91 committed:
initial commits to new mesh generator #23513

Working progress on MeshGenerator

MengnanLi91 committed:
Add test case and doco

Update MeshGenerator.C

MengnanLi91 committed:
Basic function of overlay mesh

Update OverlappingMeshGenerator.md

Update OverlayMeshGenerator.C

MengnanLi91 committed:
Modify doco

MengnanLi91 committed:
Overlay mesh generator test

Build test for overlay_mesh_generator(in progress)

MengnanLi91 committed:
Address review: improve doco and test

Co-Authored-By: Logan Harbour <loganhharbour@gmail.com>

Address review: exodus test should work for both replicated and distributed mesh

Co-authored-by: Logan Harbour <loganhharbour@gmail.com>

MengnanLi91 committed:
Modify OverlayMeshGenerator due to Mesh Generator System updates

MengnanLi91 committed:
Add elemTypes() to reduce code duplication

MengnanLi91 committed:
Address review: use applySpecificParams, params += and improve doco

Co-Authored-By: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>
Co-Authored-By: roystgnr <2367346+roystgnr@users.noreply.github.com>

Fix typo in OverlayMeshGenerator.md

MengnanLi91 committed:
Two-way elems mapping between overlay and main mesh #22493

Co-Authored-By: Logan Harbour <loganhharbour@gmail.com>

MengnanLi91 committed:
Only include boundary elems of the main mesh

MengnanLi91 committed:
Fix translation vector for overlay meshgenerator

MengnanLi91 committed:
Change find_intersection_elems from test to normal object

MengnanLi91 committed:
Add doco and comments, correct typo in test

moosebuild commented:
Job [Conda (Ubuntu)](https://civet.inl.gov/job/1469108/) on 6ac2412 : invalidated by @MengnanLi91

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1469079/) on 6ac2412 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/24098/site)

This comment will be updated on new commits.

MengnanLi91 committed:
Merge IntersectionElemsHelper to RaytracingOverlayMeshMapping, add parallel test

Co-Authored-By: Logan Harbour <loganhharbour@gmail.com>

Remove the unused variable, update the doco name in test

MengnanLi91 committed:
Add processor_id_type

MengnanLi91 committed:
Disable distributed mesh in serialize test

MengnanLi91 committed:
Add mesh parallel_type for serialize test

MengnanLi91 committed:
Use GeneratedMeshGenerator for OverlayMeshGenerator

Overlay mapping doesn't consider distributed mesh for now. We will go back to DistributedRectilinearMeshGenerator when the serialized mapping is done.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1483983/) on d33d8c0 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/24098/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/24098/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format e84297e188dde47ffd658f7e933959f4d203c443`

MengnanLi91 committed:
Clean format for precheck

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1484252/) on 0c48692 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>e84297</th>
      <th colspan=4>#24098 0c4869</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.15%</td>
      <td>85.15%</td>
      <td>+0.00%</td>
      <td>97.83%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>86882</td>
      <td>86922</td>
      <td>+40</td>
      <td>45</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15155</td>
      <td>15157</td>
      <td>+2</td>
      <td>1</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/24098/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/24098/coverage/framework)


## Modules coverage

### Ray tracing

<table>
  <thead>
    <tr>
      <th></th>
      <th>e84297</th>
      <th colspan=4>#24098 0c4869</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>93.77%</td>
      <td>93.65%</td>
      <td>-0.12%</td>
      <td>87.78%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>4080</td>
      <td>4159</td>
      <td>+79</td>
      <td>79</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>271</td>
      <td>282</td>
      <td>+11</td>
      <td>11</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/24098/coverage_diff/ray_tracing)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/24098/coverage/ray_tracing)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/24098/coverage/xfem)

</details>

## Warnings

- `ray_tracing` new line coverage rate 87.78% is less than the suggested 90.0%



This comment will be updated on new commits.

MengnanLi91 committed:
Add a class description for RayTracingOverlayMeshMapping

MengnanLi91 committed:
Add 3D test for overlay mapping

MengnanLi91 committed:
Add details to tests

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1499924/) on df24b79 wanted to post the following:


The following coverage requirement(s) failed:
- Failed to generate `chemical_reactions` coverage rate (required: 92.0%)
- Failed to generate `contact` coverage rate (required: 87.0%)
- Failed to generate `electromagnetics` coverage rate (required: 95.0%)
- Failed to generate `external_petsc_solver` coverage rate (required: 85.0%)
- Failed to generate `fluid_properties` coverage rate (required: 84.0%)
- Failed to generate `fsi` coverage rate (required: 90.0%)
- Failed to generate `functional_expansion_tools` coverage rate (required: 82.0%)
- Failed to generate `geochemistry` coverage rate (required: 96.0%)
- Failed to generate `heat_conduction` coverage rate (required: 87.0%)
- Failed to generate `level_set` coverage rate (required: 88.0%)
- Failed to generate `misc` coverage rate (required: 34.0%)
- Failed to generate `navier_stokes` coverage rate (required: 77.0%)
- Failed to generate `optimization` coverage rate (required: 86.0%)
- Failed to generate `peridynamics` coverage rate (required: 78.0%)
- Failed to generate `phase_field` coverage rate (required: 85.0%)
- Failed to generate `porous_flow` coverage rate (required: 95.0%)
- Failed to generate `ray_tracing` coverage rate (required: 93.0%)
- Failed to generate `rdg` coverage rate (required: 64.0%)
- Failed to generate `reactor` coverage rate (required: 90.0%)
- Failed to generate `scalar_transport` coverage rate (required: 90.0%)
- Failed to generate `solid_properties` coverage rate (required: 83.0%)
- Failed to generate `stochastic_tools` coverage rate (required: 88.0%)
- Failed to generate `tensor_mechanics` coverage rate (required: 84.0%)
- Failed to generate `thermal_hydraulics` coverage rate (required: 90.0%)
- Failed to generate `xfem` coverage rate (required: 80.0%)

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1499924/) on df24b79 : invalidated by @MengnanLi91

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1499924/) on df24b79 wanted to post the following:


The following coverage requirement(s) failed:
- Failed to generate `chemical_reactions` coverage rate (required: 92.0%)
- Failed to generate `contact` coverage rate (required: 87.0%)
- Failed to generate `electromagnetics` coverage rate (required: 95.0%)
- Failed to generate `external_petsc_solver` coverage rate (required: 85.0%)
- Failed to generate `fluid_properties` coverage rate (required: 84.0%)
- Failed to generate `fsi` coverage rate (required: 90.0%)
- Failed to generate `functional_expansion_tools` coverage rate (required: 82.0%)
- Failed to generate `geochemistry` coverage rate (required: 96.0%)
- Failed to generate `heat_conduction` coverage rate (required: 87.0%)
- Failed to generate `level_set` coverage rate (required: 88.0%)
- Failed to generate `misc` coverage rate (required: 34.0%)
- Failed to generate `navier_stokes` coverage rate (required: 77.0%)
- Failed to generate `optimization` coverage rate (required: 86.0%)
- Failed to generate `peridynamics` coverage rate (required: 78.0%)
- Failed to generate `phase_field` coverage rate (required: 85.0%)
- Failed to generate `porous_flow` coverage rate (required: 95.0%)
- Failed to generate `ray_tracing` coverage rate (required: 93.0%)
- Failed to generate `rdg` coverage rate (required: 64.0%)
- Failed to generate `reactor` coverage rate (required: 90.0%)
- Failed to generate `scalar_transport` coverage rate (required: 90.0%)
- Failed to generate `solid_properties` coverage rate (required: 83.0%)
- Failed to generate `stochastic_tools` coverage rate (required: 88.0%)
- Failed to generate `tensor_mechanics` coverage rate (required: 84.0%)
- Failed to generate `thermal_hydraulics` coverage rate (required: 90.0%)
- Failed to generate `xfem` coverage rate (required: 80.0%)

GiudGiud commented:
Content moved to https://github.com/idaholab/moose/issues/22493
